src/readthedocs-config.js

```diff
@@ -59,63 +61,80 @@ export function getReadTheDocsConfig(sendUrlParam) {
     url = "/_/readthedocs-addons.json";
   }
 
-  return fetch(url, {
-    method: "GET",
-  })
-    .then((response) => {
-      if (!response.ok) {
-        throw "Error parsing configuration data";
-      }
-      return response.json();
+  return url;
+}
+
+/**
+ * Load Read the Docs configuration from API endpoint.
+ *
+ */
+export function getReadTheDocsConfig(sendUrlParam) {
+  return new Promise((resolve, reject) => {
+    let dataUser;
+    const defaultApiUrl = _getApiUrl(sendUrlParam);
+
+    fetch(defaultApiUrl, {
+      method: "GET",
     })
-    .then((data) => {
-      // We force the user to define the `<meta>` tag to be able to use Read the Docs data directly.
-      // This is to keep forward/backward compatibility without breaking integrations.
-      const metadataAddonsAPIVersion = _getMetadataAddonsAPIVersion();
-      if (metadataAddonsAPIVersion !== undefined) {
-        if (metadataAddonsAPIVersion !== data.api_version) {
-          // When the API scheme version returned doesn't match the one defined via `<meta>` tag by the user,
-          // we perform another request to get the Read the Docs response in the structure
-          // that's supported by the user and dispatch a custom event letting them know
-          // this data is ready to be consumed under `event.detail`.
-
-          url =
-            ADDONS_API_ENDPOINT +
-            new URLSearchParams({
-              url: window.location.href,
-              "client-version": CLIENT_VERSION,
-              "api-version": metadataAddonsAPIVersion,
-            });
+      .then((response) => {
+        if (!response.ok) {
+          reject("Error hitting addons API endpoint");
+        }
+        // Use the addons API data response as `dataUser`
+        dataUser = response.json();
+        return dataUser;
+      })
+      .then((data) => {
+        // Create a new Promise here to handle the user request in a different async task.
+        // This allows us to start executing our integration independently from the use one.
+        new Promise((resolve, reject) => {
```

Comment:
Is this `Promise` instance needed? You are already in an async function at this point, and should be able to chain into the next `fetch` call from what I can tell.

Reply:
I did this on purpose to create a different task so it gets executed async and don't block the current one. This is to avoid hitting the API twice _before_ executing our own addons. Does this work as I expect?

Reply:
Ah gotcha, yeah I think I see what you are going for here now. This seems good then, this is the right approach.

Reply:
I don't think it should matter much, but the argument names here are overlapping with the parent Promise. To avoid unexpected errors, these could be explicit:

```suggestion
        new Promise((resolveUser, rejectUser) => {
```